packages/sync-actions/src/utils/copy-empty-array-props.js

```diff
@@ -1,30 +1,36 @@
+import isNil from 'lodash.isnil'
 /**
  * @function copyEmptyArrayProps
  * @description Takes two objects and if there are Array props in oldObj which doesn't exist in newObj, then copy it with an empty value.
  * @param {Object} oldObj
  * @param {Object} newObj
  * @returns {Array} Ordered Array [oldObj, newObj]
  */
-export default function copyEmptyArrayProps(oldObj, newObj) {
-  const newObjWithFixedEmptyArray = Object.entries(oldObj).reduce(
-    (acc, [key, value]) => {
+export default function copyEmptyArrayProps(oldObj = {}, newObj = {}) {
+  const nextObjectWithEmptyArray = Object.entries(oldObj).reduce(
+    (nextObject, [key, value]) => {
       const isArray = Array.isArray(value)
       const newObjectValueIsUndefined = newObj[key] === undefined
 
       if (isArray && newObjectValueIsUndefined) {
-        return { ...acc, [key]: [] }
+        return { ...nextObject, [key]: [] }
       }
 
-      if (!isArray && typeof value === 'object' && newObj[key]) {
+      if (
+        !isNil(value) &&
+        !isArray &&
+        typeof value === 'object' &&
+        !isNil(newObj[key])
+      ) {
         // recursion, so we could check for nested objects
-        const returnedNewObjCopy = copyEmptyArrayProps(value, newObj[key])[1]
-        return { ...acc, [key]: returnedNewObjCopy }
+        const [, returnedNewObjCopy] = copyEmptyArrayProps(value, newObj[key])
+        return { ...nextObject, [key]: returnedNewObjCopy }
       }
 
-      return acc
+      return nextObject
     },
     { ...newObj }
   )
 
-  return [oldObj, newObjWithFixedEmptyArray]
+  return [oldObj, nextObjectWithEmptyArray]
```

Comment:
and since the consumer expects and object, what we essentially do is forward back what was given to us